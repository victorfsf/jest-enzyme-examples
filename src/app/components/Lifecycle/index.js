import Lifecycle from './Lifecycle';

export default Lifecycle;
