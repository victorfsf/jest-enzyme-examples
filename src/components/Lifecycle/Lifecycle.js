/* eslint-disable no-console */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Tag from 'components/Tag';

class Lifecycle extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleLog('constructor', props);
  }

  static getDerivedStateFromProps(props) {
    console.log(`${props.type} getDerivedStateFromProps`);
    return {};
  }

  componentDidMount() {
    this.handleLog('componentDidMount');
  }

  shouldComponentUpdate() {
    this.handleLog('shouldComponentUpdate');
    return true;
  }

  componentDidUpdate() {
    this.handleLog('componentDidUpdate');
  }

  componentWillUnmount() {
    this.handleLog('componentWillUnmount');
  }

  getSnapshotBeforeUpdate() {
    this.handleLog('getSnapshotBeforeUpdate');
    return null;
  }

  handleLog(method, props = this.props) {
    const { type } = props;
    console.log(`${type} ${method}`);
  }

  render() {
    const { type } = this.props;
    this.handleLog('render');
    return <Tag name={type} />;
  }
}

Lifecycle.propTypes = {
  type: PropTypes.string.isRequired,
};

export default Lifecycle;
