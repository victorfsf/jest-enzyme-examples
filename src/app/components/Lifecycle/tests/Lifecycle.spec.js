import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Lifecycle from 'app/components/Lifecycle';

describe('Lifecycle', () => {
  /* SHALLOW */

  test('shallow', () => {
    const wrapper = shallow(<Lifecycle type="shallow" />);
    expect(wrapper).toMatchSnapshot();
    // Shallow lets you "dive" into the component
    expect(wrapper.dive()).toMatchSnapshot();

    wrapper.setProps({
      type: 'shallow (setProps)',
    });
    wrapper.unmount();
  });

  /* MOUNT */

  test('mount', () => {
    const wrapper = mount(<Lifecycle type="mount" />);
    expect(wrapper).toMatchSnapshot();
    // Mount renders everything, so you can't use dive
    expect(wrapper.dive).toBeUndefined();

    wrapper.setProps({
      type: 'mount (setProps)',
    });
    wrapper.unmount();
  });

  /* RENDER */

  test('render', () => {
    const wrapper = render(<Lifecycle type="render" />);
    expect(wrapper).toMatchSnapshot();
    // Render renders everything, so you can't use dive
    expect(wrapper.dive).toBeUndefined();
    // Render doesn't care about the lifecycle of the component
    expect(wrapper.setProps).toBeUndefined();
    expect(wrapper.unmount).toBeUndefined();
  });
});
