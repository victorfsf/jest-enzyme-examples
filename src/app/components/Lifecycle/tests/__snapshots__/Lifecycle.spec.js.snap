// Jest Snapshot v1, https://goo.gl/fbAQLP

/* SHALLOW */

exports[`Lifecycle shallow 1`] = `
<Tag
  name="shallow"
/>
`;

exports[`Lifecycle shallow 2`] = `
<button
  className="tag clickable"
  onClick={[Function]}
  type="button"
>
  shallow
</button>
`;

/* MOUNT */

exports[`Lifecycle mount 1`] = `
<Lifecycle
  type="mount"
>
  <Tag
    name="mount"
  >
    <button
      className="tag clickable"
      onClick={[Function]}
      type="button"
    >
      mount
    </button>
  </Tag>
</Lifecycle>
`;

/* RENDER */

exports[`Lifecycle render 1`] = `
<button
  class="tag clickable"
  type="button"
>
  render
</button>
`;
